mfs/dir.go

```diff
@@ -372,6 +372,30 @@ func (d *Directory) AddChild(name string, nd ipld.Node) error {
 	return nil
 }
 
+// AddChildUnixFSDir adds a child to the inner UnixFS
+// directory and transitions to a HAMT sharding if needed.
+// TODO: Rename. Actually the real confusion is that
+// this structure has a `AddChild` function that calls
+// another `AddChild` function of its inner dir.
+func (d *Directory) AddChildUnixFSDir(name string, nd ipld.Node) error {
```

Comment:
@whyrusleeping What do you think about moving the HAMT transition to the MFS layer?

Reply:
I guess the answer to that question depends on when we will need to do this, but I think its probably fine to have it in mfs